src/Picker.jsx

```diff
@@ -84,8 +85,12 @@ const Picker = createReactClass({
 
   componentDidUpdate(_, prevState) {
     if (!prevState.open && this.state.open) {
-      // setTimeout is for making sure saveCalendarRef happen before focusCalendar
-      this.focusTimeout = setTimeout(this.focusCalendar, 0, this);
+      // setTimeout is for making sure saveRef happen before focus
+      if (!this.props.calendar.props.showDateInput) {
+        this.focusTimeout = setTimeout(this.focusCalendar, 0, this);
+      } else {
+        this.focusTimeout = setTimeout(this.focusDateInput, 0, this);
```

Comment:
If we focus directly in date input, then users cannot change date with `up` `down` and so on.